api/parser_handler.go

```diff
@@ -0,0 +1,59 @@
+package api
+
+import (
+	"net/http"
+
+	"github.com/test-network-function/collector/storage"
+	"github.com/test-network-function/collector/util"
+)
+
+func ParserHandler(w http.ResponseWriter, r *http.Request, storage *storage.MySqlStorage) {
+	db := storage.MySql
+	defer db.Close()
+
+	// 1. Validate the request (includes validation of the claim file format)
+	claimFileMap, params, isValid := validatePostRequest(w, r)
+	if !isValid {
+		return
+	}
+
+	// 2. Validate claim results, and forms the input for db insert queries
+	isValid, claimResults := verifyClaimResultInJson(w, claimFileMap)
```

Comment:
What is the purpose of the separation of step 1 and 2? The validation of the claim results is also apart of the validation of the file's format.
Maybe it will be better calling `verifyClaimResultInJson(w, claimFileMap)` from inside  `validatePostRequest`?

Reply:
Thanks for pointing this out. We need only claimResults for database insertion.